### ISSUE TITLE
Chat Code Changes are not applied completely leading to VS Code needing to be restarted

### ISSUE THREAD

**Issue by mattjhussey** [microsoft/vscode-copilot-release#2433]
- Copilot Chat Extension Version: 0.23.2024102903 and also 0.22.2
- VS Code Version: 1.95.2
- OS Version: Windows 11 Pro - 10.0.22631 Build 22631
- Logs: Attached

Steps to Reproduce:

1. Add a file to the context
2. Ask a question to update the file (seems worse when updating the whole file)
3. Click to apply the changes to the file - This runs through the file making the changes
4. Expect to see "accept changes" but it is not above the editor as a tick, nor anywhere as differences in the file.
5. Any further attempts to apply changes, whether the same change or a new question result in a pop up "Another code change is being previewed. Please apply or discard the pending changes first".

Choice of LLM to use doesn't seem to matter.
The only way I've found to fix this is a VS Code restart, which is frustrating when it's happening almost every time![Image](https://github.com/user-attachments/assets/a7877c75-0c46-41cf-8356-a9606c33bb00)
[copilot.chat.log](https://github.com/user-attachments/files/17679018/copilot.chat.log)
[copilot.log](https://github.com/user-attachments/files/17679017/copilot.log)

**Comment by AnilGupte** [comment]
I am having the same problem but I am able to update the code by reopening the file instead of shutting down the editor.

**Comment by linus-ahlemeyer** [comment]
I can't fix this, even after restarting the IDE and the whole system it doesn't go away. Anyone any other idea?

**Comment by mbraun22** [comment]
I have this issue on my mac as well. Even after manually saving the file copilot wont make mew changes. I’m using cline extension as well.

**Comment by linus-ahlemeyer** [comment]
> I have this issue on my mac as well. Even after manually saving the file copilot wont make mew changes. I’m using cline extension as well.

Yep, dug some more, I have it happen only on Linux. On Windows the situation can be solved mostly.

**Comment by houba80** [comment]
review the changing part you will find at the top of the changed part apply changes

**Comment by aeschli** [comment]
The UI got reimplemented and now is the same as for Edits.
Closing the issue.

**Comment by garyfrey0** [comment]
> review the changing part you will find at the top of the changed part apply changes

I never would have found that on my own if I wasn't looking for exactly that just saying
